core/lib/src/response/response.rs

```diff
@@ -522,46 +573,57 @@ impl<'r> ResponseBuilder<'r> {
     /// let custom_values: Vec<_> = response.headers().get("X-Custom").collect();
     /// assert_eq!(custom_values, vec!["value 2", "value 3", "value 1"]);
     /// # }
+    /// 
+    /// # })
     /// ```
     #[inline(always)]
     pub fn join(&mut self, other: Response<'r>) -> &mut ResponseBuilder<'r> {
         self.response.join(other);
         self
     }
 
-    /// Retrieve the built `Response`.
-    ///
-    /// # Example
-    ///
-    /// ```rust
-    /// use rocket::Response;
-    ///
-    /// # #[allow(unused_variables)]
-    /// let response = Response::build()
-    ///     // build the response
-    ///     .finalize();
-    /// ```
-    #[inline(always)]
-    pub fn finalize(&mut self) -> Response<'r> {
-        std::mem::replace(&mut self.response, Response::new())
-    }
-
     /// Retrieve the built `Response` wrapped in `Ok`.
     ///
     /// # Example
     ///
     /// ```rust
     /// use rocket::Response;
+    /// 
+    /// # rocket::async_test(async {
     ///
     /// let response: Result<Response, ()> = Response::build()
     ///     // build the response
-    ///     .ok();
+    ///     .ok().await;
     ///
     /// assert!(response.is_ok());
+    /// 
+    /// # })
     /// ```
     #[inline(always)]
-    pub fn ok<T>(&mut self) -> Result<Response<'r>, T> {
-        Ok(self.finalize())
+    pub async fn ok<E>(&mut self) -> Result<Response<'r>, E> {
```

Comment:
I'm tempted but not quite convinced to make `ok()` return a `Pin<Box<dyn Future<Output=Result<Response<'r>, E>>>>` instead. That would allow the common case of `Box::pin(async move { /* build */.ok().await })` to become just `/* build */.ok()` again.